### PR TITLE
Let's review pending changes

### PR DESCRIPTION
**Pull request by belyaev-mikhail**

### CODE REVIEW

File: docs/src/md/kotlin.core/cdfa.md [hunk 1]
```diff
@@ -1338,6 +1338,242 @@ At line 7 there is another compile-time error when both properties are used, as
 
 See the [corresponding section][Smart casts] for details.
 
+#### Function contracts
+
+> Note: as of Kotlin 1.4.0, contracts for user-defined functions are an experimental feature and, thus, not described here
```
Comment (belyaev-mikhail) [inline docs/src/md/kotlin.core/cdfa.md]: 1.5.0

File: docs/src/md/kotlin.core/coroutines.md [hunk 2]
```diff
@@ -28,9 +28,12 @@ The main source of suspension points are calls to other suspending functions whi
 
 Non-suspending functions may not call suspending functions directly, as they do not support suspension points.
 Suspending functions may call non-suspending functions without any limitations; such calls do not create suspension points.
+This restriction is also known as ["function colouring"](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/).
 
 > Important: an exception to this rule are non-suspending inlined lambda parameters: if the higher-order function invoking such a lambda is called from a suspending function, this lambda is allowed to also have suspension points and call other suspending functions.
 
+TODO(Cover other possible exceptions w.r.t. `inline`/`noinline`/`crossinline` combinations)
```
Comment (belyaev-mikhail) [inline docs/src/md/kotlin.core/coroutines.md:35]: um, there aren't any afaik?

File: docs/src/md/kotlin.core/coroutines.md [hunk 3]
```diff
@@ -58,4 +61,199 @@ One of the ways of starting suspending function from a non-suspending context is
...
+```
+
+When the cached `intercepted` continuation is no longer needed, it is released using `ContinuationInterceptor.releaseInterceptedContinuation(...)`.
+
+> Note: this machinery is performed "behind-the-scenes" by the coroutine framework implementation.
+
+#### Coroutine intrinsics
+
+Accessing the low-level continuations is performed using a limited number of built-in intrinsic functions, which form the complete coroutine API.
+The rest of asynchronous programming support is provided as a Kotlin library.
```
Comment (belyaev-mikhail) [inline docs/src/md/kotlin.core/coroutines.md]: Name of the library?

File: docs/src/md/kotlin.core/coroutines.md [hunk 4]
```diff
@@ -58,4 +61,199 @@ One of the ways of starting suspending function from a non-suspending context is
...
+The complete built-in API for working with coroutines is shown below.
```
Comment (belyaev-mikhail) [inline docs/src/md/kotlin.core/coroutines.md]: package?

File: docs/src/md/kotlin.core/inheritance.md [hunk 5]
```diff
@@ -31,13 +31,15 @@ A class declared `abstract` cannot be instantiated, i.e., an object of this clas
...
-#### Sealed classes
+#### Sealed classes and interfaces
 
-A class may be declared `sealed`, making it special from the inheritance point-of-view.
+A class or interface (but not a [functional interface][Functional interface declaration]) may be declared `sealed`, making it special from the inheritance point-of-view.
 
 - A `sealed` class is implicitly `abstract` (and these two modifiers are exclusive);
-- It can only be inherited from by class and object types declared in the same file (including class and object types declared as nested classes for this class, but not nested classes in other classes);
-- It allows for exhaustiveness checking of [when expressions][When expressions] for values of this type.
+- A `selaed` class or interface can only be inherited from by types declared in the same package and in the same [module][Modules], and which have a fully-qualified name (meaning local and anonymous types cannon be inherited from `sealed` types);
```
Comment (belyaev-mikhail) [inline docs/src/md/kotlin.core/inheritance.md]: typo

File: docs/src/md/kotlin.core/builtins.md [hunk 6]
```diff
@@ -197,13 +200,18 @@ It is the type of the result of [string interpolation][String interpolation expr
   public override final fun hashCode(): Int
   ```
 
+    These methods are defined to their default behaviour: only the same entry of an enum class is equal to itself and no other object.
```
Comment (ice-phoenix) [inline docs/src/md/kotlin.core/builtins.md]: methods -> member functions

File: docs/src/md/kotlin.core/cdfa.md [hunk 7]
```diff
@@ -1338,6 +1338,242 @@ At line 7 there is another compile-time error when both properties are used, as
...
+- Calls-in-place effect for a function-type parameter of the function;
+- Returns-implies-condition effect for a boolean parameter of the function;
+- Particular implementations may introduce other types of effects.
+
+**Calls-in-place** effect of function $F$ for a function-type parameter $P$ specifies that for every call of $F$ parameter $P$ will be also invoked as a function.
+This effect may also have one of the three invocation types: 
+
+- *At-least-once*, meaning that the $P$ will be invoked at least once;
+- *Exactly-once*, meaning that the $P$ will be invoked exactly once;
+- *At-most-once*, meaning that the $P$ will be invoked at most once.
```
Comment (ice-phoenix) [inline docs/src/md/kotlin.core/cdfa.md]: Drop the "the"?

File: docs/src/md/kotlin.core/cdfa.md [hunk 8]
```diff
@@ -1338,6 +1338,242 @@ At line 7 there is another compile-time error when both properties are used, as
...
+|   $result = f(...,$N,..)   |      |                   |
+|                            |      |                   |
++-------------+--------------+      |                   |
+              |                     v                   v
+              +---------------------+-------------------+
+              |
+              v
+```
+
+This allows the control-flow information to be extracted from lambda expression according to the policy of their invocation.
```
Comment (ice-phoenix) [inline docs/src/md/kotlin.core/cdfa.md]: ... its invocation ...

**Review by ice-phoenix — APPROVED** [review]

**Pull request merged**